tests/php/RestApi/StoreApi/Controllers/CartItems.php

```diff
@@ -229,4 +238,26 @@ public function test_prepare_item_for_response() {
 		$this->assertArrayHasKey( 'totals', $response->get_data() );
 		$this->assertArrayHasKey( 'variation', $response->get_data() );
 	}
+
+	/**
+	 * Test schema matches responses.
+	 *
+	 * Tests schema of both products in cart to cover as much schema as possible.
+	 */
+	public function test_schema_matches_response() {
+		$cart       = wc()->cart->get_cart();
+		$controller = new \Automattic\WooCommerce\Blocks\RestApi\StoreApi\Controllers\CartItems();
+		$schema     = $controller->get_item_schema();
+		$validate   = new ValidateSchema( $schema );
+
+		// Simple product.
+		$response = $controller->prepare_item_for_response( current( $cart ), [] );
+		$diff     = $validate->get_diff_from_object( $response->get_data() );
+		$this->assertEquals( [ 'variation' ], $diff, print_r( $diff, true ) );
+
+		// Variable product.
+		$response = $controller->prepare_item_for_response( end( $cart ), [] );
+		$diff     = $validate->get_diff_from_object( $response->get_data() );
+		$this->assertEquals( [ 'images' ], $diff, print_r( $diff, true ) );
```

Comment:
🤔 , why are we expecting there should be a diff result for `variation` and `images` here?  Unless I'm misunderstanding, it seems that `setUp` is adding some variations and images for products added to the cart?

Reply:
There are 2 products in the cart. 1 is simple (has no variations, so I expect it to be missing) and one has no image (again, expected so this asserts only that is missing).